### PR TITLE
Add chat mode for finetuned models using Adapter.

### PR DESCRIPTION
**Pull request by shuwang127**
To chat with the finetuned model interactively using:
  python chat/adapter.py
The default parameters:
    - model path: out/adapter/alpaca/lit_model_adapter_finetuned.pth
    - model type: stabilityai/stablelm-base-alpha-3b
    - finetune data: alpaca
    - finetune tech: Adapter
Test success on NVIDIA DGX-A100.

### CODE REVIEW

**Comment by shuwang127** [comment]
This is a separate script added to the ./chat/ folder, won't affect the other script logic.

File: chat/adapter.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+from typing import Literal
+
+import lightning as L
+import torch
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from generate.base import generate
```
Comment (carmocca) [inline chat/adapter.py:15]: You are using the generation code from the generation/base.py script, instead of chat/base.py.

Is that intended?

**Review by carmocca — COMMENTED** [review]
I resolved conflicts. Just one question

File: chat/adapter.py (hunk 1 above)
Comment (shuwang127) [reply chat/adapter.py:15]: When I use `from chat.base import generate`

There will be an issue on:

File "<path>/lit-gpt/lit_parrot/tokenizer.py", line 50, in decode
    tokens = [tensor.item()] if tensor.ndim == 0 else tensor.tolist()
AttributeError: 'generator' object has no attribute 'ndim'

But `from generate.base import generate` will be good.

Comment (carmocca) [reply chat/adapter.py:15]: I see. Your current implementation won't work. chat.base.generate returns a generator of tokens whereas your current code expects a tensor of tokens. Please try out the `chat/base.py` script to understand how it works.

If the adapter checkpoints will always start with `### Response:`, we can simply ignore the tokens associated to that prefix.

**Review by carmocca — CHANGES_REQUESTED** [review]
The current implementation "fakes" streaming support, because the tokens are generated all at once but then printed one by one. We should mimic the `chat/base.py` script

**Pull request closed without merging**